pkg/registry/apps/statefulset/strategy.go

```diff
@@ -85,6 +87,10 @@ func (statefulSetStrategy) PrepareForCreate(ctx context.Context, obj runtime.Obj
 
 	statefulSet.Generation = 1
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetAutoDeletePVC) {
```

Comment:
This should really be in a statefulset.DropDisabledFields method (i guess it could be in this package).  We will have others.

Reply:
Done